blocks/library/cover-image/index.js

```diff
@@ -94,121 +100,142 @@ export const settings = {
 		}
 	},
 
-	edit( { attributes, setAttributes, isSelected, className } ) {
-		const { url, title, align, contentAlign, id, hasParallax, dimRatio } = attributes;
-		const updateAlignment = ( nextAlign ) => setAttributes( { align: nextAlign } );
-		const onSelectImage = ( media ) => setAttributes( { url: media.url, id: media.id } );
-		const toggleParallax = () => setAttributes( { hasParallax: ! hasParallax } );
-		const setDimRatio = ( ratio ) => setAttributes( { dimRatio: ratio } );
+	edit: withSelect( ( select, ownProps ) => ( {
+		hasCallAction: !! select( 'core/editor' ).getBlocks( ownProps.id ).length,
```

Comment:
`getBlocks` can be an expensive operation since it requires transforming the blocks into its object form, including inner blocks and meta attributes. In this case `getBlockOrder` would be much cheaper, or even `getBlockCount` if all you're trying to do is see if there exists an inner block.